### PR TITLE
fix(entwine): change transparency settings

### LINKED ISSUE #2323: [BUG] Transparent individual pointcloud classes don't get rendered as transparent
## Context
I am using classified pointclouds provided as 3D Tiles. 
I am setting a custom classificationScheme to choose color and opacity for each individual pointcloud class.
Setting colors works fine. Reducing opacity will however fade the point color to white instead of making it transparent.
On the screenshot bellow the buildings are excpected to look half-transparent red but appear pink and opaque instead.

image

## Steps to Reproduce

1. Render a classified point cloud 3D Tile source (https://raw.githubusercontent.com/iTowns/iTowns2-sample-data/master/pointclouds/pnts-sete-2021-0756_6256/tileset.json) 
2. Set a custom classification scheme with a transparent class (here buildings are set red and half-transparent and the rest is black) : 
```
layer.classification = {
                    6: { visible: true, color: new THREE.Color(1, 0, 0), opacity: 0.5 },
                    DEFAULT: { visible: true, color: new THREE.Color(0, 0, 0), opacity: 1 }
}
```

## Expected Behavior
The classified points should be rendered as transparent.

## Actual Behavior
The  classified point are getting whiter. 

## Possible Cause/Fix/Solution
It seems that the point material transparent property is not set correctly. Thus it renders the color alpha as color intensity instead of transparency. This happens because the layer opacity is set to 1. Even if we try to set the whole layer material transparent, the transparent property getter overwrites the value depending on the opacity. See here : 

https://github.com/iTowns/itowns/blob/48fb9dfd3a6b420dc1e121971eb2a6a3a2efb286/src/Layer/ReferencingLayerProperties.js#L48-L56

### PR DESCRIPTION
**Pull request by ftoromanoff**
Fix for issue #2323.

This issue point out a bug with the rendering of point cloud where points should be rendered with differentes values of opacity.

### CODE REVIEW

**Comment by airnez** [comment]
Do you think it will fix [this issue (#2323)](https://github.com/iTowns/itowns/issues/2323) ?

**Comment by ftoromanoff** [comment]
> Do you think it will fix [this issue (#2323)](https://github.com/iTowns/itowns/issues/2323) ?

Currently, I only have look up for a fix for PointCloudLayer (entwine  Potree Layers). I will try to continue toward 3DTilesLayer

File: examples/3dtiles_pointcloud.html [hunk 1]
```diff
@@ -58,14 +58,19 @@
...
                 sseThreshold: 5,
                 pntsMode: itowns.PNTS_MODE.CLASSIFICATION,
                 pntsShape : itowns.PNTS_SHAPE.CIRCLE,
-                source: new itowns.C3DTilesSource({
-                    url: 'https://raw.githubusercontent.com/iTowns/iTowns2-sample-data/master/pointclouds/pnts-sete-2021-0756_6256/tileset.json',
-                }),
+                source: $3dTilesSource,
+                classification: {
+                    6: { visible: true, color: new itowns.THREE.Color(1, 0, 0), opacity: 0.1 },
+                    DEFAULT: { visible: true, color: new itowns.THREE.Color(0, 0, 0), opacity: 1 }
```
Comment (ftoromanoff) [inline examples/3dtiles_pointcloud.html]: A supp

**Comment by airnez** [comment]
> > Do you think it will fix [this issue (#2323)](https://github.com/iTowns/itowns/issues/2323) ?
> 
> Currently, I only have look up for a fix for PointCloudLayer (entwine Potree Layers). I will try to continue toward 3DTilesLayer

Understood. Thank you for your answer 👍

**Comment by AnthonyGlt** [comment]
Thank you @ftoromanoff for this MR !
Looks like it solves the transparency issue :+1: 
It also exposed the fact that the 3dtiles is a PointCloud (or contains points) with hasPnts. I like this idea.
Looks good to me !

I have an interrogation, it's not really related to this MR but it's about PointsMaterial.js

https://github.com/iTowns/itowns/blob/0e822cb268ad702df52020e58a1029ac21472b03/src/Renderer/PointsMaterial.js#L49-L80

WHat's the point of having a DEFAULT object in the ClassificationScheme & in the DiscreteScheme ?
Couldn't we put the DiscreteScheme directly in the ClassificationScheme ? We could access it with
`ClassificationScheme.DISCRETE`

```
export const ClassificationScheme = {
    DEFAULT: {
        0: { visible: true, name: 'never classified', color: new THREE.Color(0.5,  0.5,  0.5), opacity: 1.0 },
        1: { visible: true, name: 'unclassified', color: new THREE.Color(0.5,  0.5,  0.5), opacity: 1.0 },
        2: { visible: true, name: 'ground', color: new THREE.Color(0.63, 0.32, 0.18), opacity: 1.0 },
        3: { visible: true, name: 'low vegetation', color: new THREE.Color(0.0,  1.0,  0.0), opacity: 1.0 },
        4: { visible: true, name: 'medium vegetation', color: new THREE.Color(0.0,  0.8,  0.0), opacity: 1.0 },
        5: { visible: true, name: 'high vegetation', color: new THREE.Color(0.0,  0.6,  0.0), opacity: 1.0 },
        6: { visible: true, name: 'building', color: new THREE.Color(1.0,  0.66, 0.0), opacity: 1.0 },
        7: { visible: true, name: 'low point(noise)', color: new THREE.Color(1.0,  0.0,  1.0), opacity: 1.0 },
        8: { visible: true, name: 'key-point', color: new THREE.Color(1.0,  0.0,  0.0), opacity: 1.0 },
        9: { visible: true, name: 'water', color: new THREE.Color(0.0,  0.0,  1.0), opacity: 1.0 },
        10: { visible: true, name: 'rail', color: new THREE.Color(0.8,  0.8,  1.0), opacity: 1.0 },
        11: { visible: true, name: 'road Surface', color: new THREE.Color(0.4,  0.4,  0.7), opacity: 1.0 },
        12: { visible: true, name: 'overlap', color: new THREE.Color(1.0,  1.0,  0.0), opacity: 1.0 },
        DEFAULT: { visible: true, name: 'default', color: new THREE.Color(0.3, 0.6, 0.6), opacity: 1.0 },
    },
 DISCRETE : {

        0: { visible: true, name: '0', color: new THREE.Color('rgb(67, 99, 216)'), opacity: 1.0 },
        1: { visible: true, name: '1', color: new THREE.Color('rgb(60, 180, 75);'), opacity: 1.0 },
        2: { visible: true, name: '2', color: new THREE.Color('rgb(255, 255, 25)'), opacity: 1.0 },
        3: { visible: true, name: '3', color: new THREE.Color('rgb(145, 30, 180)'), opacity: 1.0 },
        4: { visible: true, name: '4', color: new THREE.Color('rgb(245, 130, 49)'), opacity: 1.0 },
        5: { visible: true, name: '5', color: new THREE.Color('rgb(230, 25, 75)'), opacity: 1.0 },
        6: { visible: true, name: '6', color: new THREE.Color('rgb(66, 212, 244)'), opacity: 1.0 },
        7: { visible: true, name: '7', color: new THREE.Color('rgb(240, 50, 230)'), opacity: 1.0 },
        DEFAULT: { visible: true, name: 'default', color: white, opacity: 1.0 },
    },
};
```
WDYT ?

**Comment by ftoromanoff** [comment]
> Thank you @ftoromanoff for this MR ! Looks like it solves the transparency issue 👍 It also exposed the fact that the 3dtiles is a PointCloud (or contains points) with hasPnts. I like this idea. Looks good to me !
> 
> I have an interrogation, it's not really related to this MR but it's about PointsMaterial.js
> 
> https://github.com/iTowns/itowns/blob/0e822cb268ad702df52020e58a1029ac21472b03/src/Renderer/PointsMaterial.js#L49-L80
> 
> WHat's the point of having a DEFAULT object in the ClassificationScheme & in the DiscreteScheme ? Couldn't we put the DiscreteScheme directly in the ClassificationScheme ? We could access it with `ClassificationScheme.DISCRETE`
> 
> ```
> export const ClassificationScheme = {
>     DEFAULT: {
>         0: { visible: true, name: 'never classified', color: new THREE.Color(0.5,  0.5,  0.5), opacity: 1.0 },
>         1: { visible: true, name: 'unclassified', color: new THREE.Color(0.5,  0.5,  0.5), opacity: 1.0 },
>         2: { visible: true, name: 'ground', color: new THREE.Color(0.63, 0.32, 0.18), opacity: 1.0 },
>         3: { visible: true, name: 'low vegetation', color: new THREE.Color(0.0,  1.0,  0.0), opacity: 1.0 },
>         4: { visible: true, name: 'medium vegetation', color: new THREE.Color(0.0,  0.8,  0.0), opacity: 1.0 },
>         5: { visible: true, name: 'high vegetation', color: new THREE.Color(0.0,  0.6,  0.0), opacity: 1.0 },
>         6: { visible: true, name: 'building', color: new THREE.Color(1.0,  0.66, 0.0), opacity: 1.0 },
>         7: { visible: true, name: 'low point(noise)', color: new THREE.Color(1.0,  0.0,  1.0), opacity: 1.0 },
>         8: { visible: true, name: 'key-point', color: new THREE.Color(1.0,  0.0,  0.0), opacity: 1.0 },
>         9: { visible: true, name: 'water', color: new THREE.Color(0.0,  0.0,  1.0), opacity: 1.0 },
>         10: { visible: true, name: 'rail', color: new THREE.Color(0.8,  0.8,  1.0), opacity: 1.0 },
>         11: { visible: true, name: 'road Surface', color: new THREE.Color(0.4,  0.4,  0.7), opacity: 1.0 },
>         12: { visible: true, name: 'overlap', color: new THREE.Color(1.0,  1.0,  0.0), opacity: 1.0 },
>         DEFAULT: { visible: true, name: 'default', color: new THREE.Color(0.3, 0.6, 0.6), opacity: 1.0 },
>     },
>  DISCRETE : {
> 
>         0: { visible: true, name: '0', color: new THREE.Color('rgb(67, 99, 216)'), opacity: 1.0 },
>         1: { visible: true, name: '1', color: new THREE.Color('rgb(60, 180, 75);'), opacity: 1.0 },
>         2: { visible: true, name: '2', color: new THREE.Color('rgb(255, 255, 25)'), opacity: 1.0 },
>         3: { visible: true, name: '3', color: new THREE.Color('rgb(145, 30, 180)'), opacity: 1.0 },
>         4: { visible: true, name: '4', color: new THREE.Color('rgb(245, 130, 49)'), opacity: 1.0 },
>         5: { visible: true, name: '5', color: new THREE.Color('rgb(230, 25, 75)'), opacity: 1.0 },
>         6: { visible: true, name: '6', color: new THREE.Color('rgb(66, 212, 244)'), opacity: 1.0 },
>         7: { visible: true, name: '7', color: new THREE.Color('rgb(240, 50, 230)'), opacity: 1.0 },
>         DEFAULT: { visible: true, name: 'default', color: white, opacity: 1.0 },
>     },
> };
> ```
> 
> WDYT ?

The discrete cheme is indeed a scheme for point classification. But ClassificationScheme is for the 'classification' point classification. That's the reason why there is 2 differentes schemes, One for the classification using point classification and an other one for all the other (return number, type of return etc...).  We can imagine that in a future (or a user) want to have different discrete scheme: one for the type of return and another one for the point source ID.

**Comment by ftoromanoff** [comment]
@AnthonyGlt Can you start a review, and validate it if you have no remarks on the code ?
Thks

**Review by AnthonyGlt — APPROVED** [review]

**Pull request merged**